library/src/main/java/com/alamkanak/weekview/WeekView.kt

```diff
@@ -0,0 +1,2525 @@
+
+        val hourLines = FloatArray(lineCount * 4)
+
+        // Clear the cache for event rectangles.
+        if (mEventRects != null) {
+            for (eventRect in mEventRects!!) {
+                eventRect.rectF = null
+            }
+        }
+
+        // Clip to paint events only.
+        canvas.clipRect(mHeaderColumnWidth, mHeaderHeight + (mHeaderRowPadding * 2).toFloat() + mHeaderMarginBottom + mTimeTextHeight / 2, width.toFloat(), height.toFloat(), Region.Op.REPLACE)
+
+        // Iterate through each day.
+        val oldFirstVisibleDay = firstVisibleDay
+        firstVisibleDay = mHomeDate!!.clone() as Calendar
+        firstVisibleDay!!.add(Calendar.DATE, -Math.round(mCurrentOrigin.x / (mWidthPerDay + mColumnGap)))
+        if (firstVisibleDay != oldFirstVisibleDay && scrollListener != null) {
+            scrollListener!!.onFirstVisibleDayChanged(firstVisibleDay!!, oldFirstVisibleDay)
+        }
+
+        if (autoLimitTime) {
+            val days = ArrayList<Calendar>()
+            for (dayNumber in leftDaysWithGaps + 1..leftDaysWithGaps + realNumberOfVisibleDays) {
+                val day = mHomeDate!!.clone() as Calendar
+                day.add(Calendar.DATE, dayNumber - 1)
+                days.add(day)
+            }
+            limitEventTime(days)
+        }
+
+        for (dayNumber in leftDaysWithGaps + 1..leftDaysWithGaps + realNumberOfVisibleDays + 1) {
+            // Check if the day is today.
+            val day = mHomeDate!!.clone() as Calendar
+            lastVisibleDay = day.clone() as Calendar
+            day.add(Calendar.DATE, dayNumber - 1)
+            lastVisibleDay!!.add(Calendar.DATE, dayNumber - 2)
+            val isToday = isSameDay(day, today)
+
+            // Don't draw days which are outside requested range
+            if (!dateIsValid(day)) {
+                continue
+            }
+
+            // Get more events if necessary. We want to store the events 3 months beforehand. Get
+            // events only when it is the first iteration of the loop.
+            if (mEventRects == null || mRefreshEvents ||
+                    dayNumber == leftDaysWithGaps + 1 && mFetchedPeriod != weekViewLoader!!.toWeekViewPeriodIndex(day).toInt() &&
+                    Math.abs(mFetchedPeriod - weekViewLoader!!.toWeekViewPeriodIndex(day)) > 0.5) {
+                getMoreEvents(day)
+                mRefreshEvents = false
+            }
+
+            // Draw background color for each day.
+            val start = if (startPixel < mHeaderColumnWidth) mHeaderColumnWidth else startPixel
+            if (mWidthPerDay + startPixel - start > 0) {
+                if (mShowDistinctPastFutureColor) {
+                    val isWeekend = day.get(Calendar.DAY_OF_WEEK) == Calendar.SATURDAY || day.get(Calendar.DAY_OF_WEEK) == Calendar.SUNDAY
+                    val pastPaint = if (isWeekend && mShowDistinctWeekendColor) mPastWeekendBackgroundPaint else mPastBackgroundPaint
+     
```

Comment:
After mCurrentOrigin.x = ...
just add the line deleted just before : 
`mStartOriginForScroll = mCurrentOrigin.x`